### ISSUE TITLE
Provide a formatted error output

### ISSUE THREAD

**Issue by sindresorhus** [sass/libsass#938]
Moved from https://github.com/sass/node-sass/issues/687

Would be useful for consumer to just be able to log out an already nicely formatted error string. Currently I have to [manually create](https://github.com/sindresorhus/grunt-sass/blob/5b0c8a231574f043e16c3d2c2e7c6b84bc1796c3/tasks/sass.js#L36) the error output in my grunt plugin.

I'm thinking something like this (but also with filename):

![image](https://cloud.githubusercontent.com/assets/170270/6185814/502fc4c2-b3a5-11e4-9072-41571c128d20.png)

*(from [`gulp-jscs`](https://github.com/jscs-dev/gulp-jscs))*

**Comment by mgreter** [comment]
We provide all the information needed to let implementers do this as they like. IMO we don't want to create such "text pointers" inside libsass, since once probably wants it in this style, while other prefer it in a completely different style.

**Comment by sindresorhus** [comment]
@mgreter I know we can implement it ourselves, but that means every consumers has to manually implement it. Would be nice if libsass came with a nice output. Consumers are still free to create their own based on the error info.

**Comment by mgreter** [comment]
OK, I guess this probably is a usefull feature and you're lucky since we have a field that we "mangle" anyway in `sass_context` (//CC @am11). We now actually provide two "messages" for errors on the context `error_text` contains the original error message and `error_message` will contain a "decorated" message, in the original ruby sass style, plus the new arrow indicator.

```
Error: invalid attribute name in attribute selector
        on line 1 of test.scss
>> foo [ { bar: baz; }
   -----^
```

Currently we have a minor mess at how we report errors, since we do not really report them in the same way as ruby sass does (and we don't test them), at some points we even give more detailed error messages. I added some code in preparation to streamline this in the PR for this issue, so at least the output format is the same, although this PR adds something that is not in ruby sass, but the implementor can still chose to get the original text to format it as he sees fit!

**Comment by am11** [comment]
:+1: 
Can you please keep the `message` member intact and add a `description` (or `detail`) to JSON for the decorator error? Actually, `message` is aligned with JavaScript's `Error` property. :)
Single word like `description` or `detail` is also appreciated because JSON's preferred style is cameCase and other languages may not appreciate having no `_` between disjoint words. :)

**Comment by mgreter** [comment]
This does not touch `json_err`, only the raw objects on the context. Not sure in what format it would make sense to pass further down, you can always get it via C-API, it's just not part of the json string. Actually the "message" on json is the same as `err_text` on context, only `err_message` on context is "mangled" by libsass (and it has been that way before, just not hat "extensive").

**Comment by am11** [comment]
Ok. Would it be possible to add `err_message` to JSON endpoint as `detail` as well? That would be really cool! This way we won't need to do extra effort in JS. :)

**Comment by xzyfer** [comment]
Could we use more descriptive members. `text` vs `message` aren't really useful. Can call it `formatted` or `fancy` (joking, not joking) ?

**Comment by am11** [comment]
+1 for `formatted`. So the structure of JSON tuple would be https://github.com/sass/node-sass#error-object + `formatted`. Note that currently we do not need to [induce any property](https://github.com/sass/node-sass/blob/2220ba96b9a20920ccfaaadc12978f40e069caf9/lib/index.js#L151) and forward the `Error` object (parsed from libsass' JSON string) as is to the downstream. So the `formatted` addition in libsass would automagically be incorporated in node-sass. (ok, I am being selfish here)

**Comment by am11** [comment]
@mgreter, node v3.0.0-beta.5 is released, that corresponds to libsass v3.2.0-beta.5. :tada: :gift: 

I was hoping it would close https://github.com/sindresorhus/grunt-sass/issues/201 without intervention of node-sass, but the formatted errors are not emitted in JSON error string. :(

Input:

```
data: 'a{b:c'
```

Output (error):

```json
{ "message": "error reading values after c",
  "column": 5,
  "line": 1,
  "file": "stdin",
  "status": 1 }
```

Expected:

```json
{ "message": "error reading values after c",
  "formatted": "error reading values after c\non line 1 of stdin\n>> a{b:c\n-------^",
  "column": 5,
  "line": 1,
  "file": "stdin",
  "status": 1 }
```

If we are suppose to parse, unwind and then repack JSON error string to implode decorated/formatted error member, then it is really an ordeal. IMO, it seems like an unnecessary overhead to capture part of error differently.
